### PR TITLE
Add notify build status action

### PR DESCRIPTION
**Pull request by reakaleek**
## What does this PR do?

Creates a github action that notifies about the build status. Currently, only supports slack. May be extended in the future.

image

## Why is it important?

Unified notifications.

## Related issues
pipeline migration

### CODE REVIEW

**Review by v1v — COMMENTED** [review]
Do you think https://github.com/elastic/apm-pipeline-library/blob/main/.github/actions/slack-message/action.yml could be refactored in a way the `notify-build-status` could used it?

As far as I see the differ in the payload and how to get it.

Otherwise, it might be a good idea to [unset the slack bot token](https://github.com/elastic/apm-pipeline-library/blob/0927156640c9783fda91b230a37a4f69e95cf12e/.github/actions/slack-message/action.yml#L99-L102)

**Comment by reakaleek** [comment]
> Do you think https://github.com/elastic/apm-pipeline-library/blob/main/.github/actions/slack-message/action.yml could be refactored in a way the `notify-build-status` could used it?
> 
> As far as I see the differ in the payload and how to get it.
> 
> Otherwise, it might be a good idea to [unset the slack bot token](https://github.com/elastic/apm-pipeline-library/blob/0927156640c9783fda91b230a37a4f69e95cf12e/.github/actions/slack-message/action.yml#L99-L102)

https://github.com/elastic/apm-pipeline-library/pull/2048/commits/6e74bbdd924bfc92558fba5ae6dc6461a3dab303

File: .github/actions/notify-build-status/action.yml [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+  using: composite
+  steps:
+    - uses: hashicorp/vault-action@v2.4.2
+      with:
+        url: ${{ inputs.vaultUrl }}
+        roleId: ${{ inputs.vaultRoleId }}
+        secretId: ${{ inputs.vaultSecretId }}
+        method: approle
+        secrets: |
+          secret/observability-team/ci/stack-integartion-token integration_token | SLACK_BOT_TOKEN
```
Comment (v1v) [inline .github/actions/notify-build-status/action.yml:28]: leftover?

```suggestion
```

Comment (reakaleek) [reply .github/actions/notify-build-status/action.yml]: Oh yes, thank you for noticing: [001a698](https://github.com/elastic/apm-pipeline-library/pull/2048/commits/001a698eaa20f96f9bc3f5be53d2bb0156318cf7)

**Review by v1v — APPROVED** [review]

**Pull request merged**